### PR TITLE
Add filemergers for mergemodes

### PR DESCRIPTION
**Pull request by TheMysticMan**
Adds basic filemergers for skip and replace MergeModes.
Allow multiple Entities and Attributes in DesignModel.

### CODE REVIEW

File: Polygen.Core/Impl/Project/ProjectFile.cs [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
+        {
+        }
+
+        public ProjectFile(IProject project, string relativePath) : base(project, relativePath, false)
+        {
+        }
+
+        public TextReader OpenAsTextForReading()
+        {
+            return new StreamReader(System.IO.File.Open(this.GetSourcePath(true), FileMode.Open), Encoding.UTF8);
```
Comment (mpoyhonen) [inline Polygen.Core/Impl/Project/ProjectFile.cs:21]: Why is this added?

Comment (TheMysticMan) [reply Polygen.Core/Impl/Project/ProjectFile.cs:21]: Because of a Namespace conflict. If you want to I could move the file merge functionality to another namespace. It's in Polygen.Core.File at the moment.

**Review by mpoyhonen — APPROVED** [review]

**Comment by mpoyhonen** [comment]
I think this is ok for now. I might refactor this a bit, especially the factory implementation should not be necessary..
